.circleci/config.yml

```diff
@@ -0,0 +1,106 @@
+defaults: &defaults
+  working_directory: ~/prisoner-content-hub-backend
+  docker:
+    - image: circleci/php:7.1
+main_branch: &main_branch
+  filters:
+    branches:
+      only: main
+feature_branch: &feature_branch
+  filters:
+    branches:
+      ignore: main
+
+version: 2.1
+
+jobs:
+  build-preview:
+    <<: *defaults
+    steps:
+      - checkout
+      - setup_remote_docker:
+          docker_layer_caching: true
+      - run: sudo composer self-update
+      - restore_cache:
+          key: dependency-cache-{{ checksum "composer.lock" }}
+      - run:
+          name: Install Drupal modules
+          command: |
+            make build-deps
+      - save_cache:
+          key: dependency-cache-{{ checksum "composer.lock" }}
+          paths:
+            - ./modules/contrib
+      - run:
+          name: Build Docker image
+          command: make build
+      - run:
+          name: Push Docker image
+          command: make push-preview
+      - run:
+          name: Persist build number for deployment
+          command: |
+            mkdir -p /tmp/workspace/docker-build
+            cd /tmp/workspace/docker-build
+            echo preview > version-to-deploy.txt
+      - persist_to_workspace:
+          root: /tmp/workspace
+          paths:
+            - docker-build
+  build_drupal:
```

Comment:
My only comment is maybe call it `build_prod` to make the job names consistent as they both build Drupal 😄

Reply:
Point taken, will update 😄